### PR TITLE
Node id update command

### PR DESCRIPTION
**Pull request by Caleb-Hill**
Adds node_id import functionality to the cli update command.

### CODE REVIEW

File: libsplinter/src/node_id/store/mod.rs [hunk 1]
```diff
@@ -16,6 +16,7 @@
 pub mod diesel;
 /// NodeIdStore error types
```
Comment (peterschwarz) [inline libsplinter/src/node_id/store/mod.rs:17]: Not part of this PR, but this should be module doc in the error module, not a comment here.

**Review by peterschwarz — CHANGES_REQUESTED** [review]

File: libsplinter/src/node_id/store/file/mod.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+//! NodeIdStore that supports the 0.4 node_id file format.
+
+use std::fs;
+use std::path::PathBuf;
+
+use super::NodeIdStore;
+use super::NodeIdStoreError;
+
+/// The 0.4 node_id file is depricated, this exists to help migrate the node_id.
```
Comment (peterschwarz) [inline libsplinter/src/node_id/store/file/mod.rs:23]: depricate -> deprecated

But more specifically, change it to "soft-deprecated", as Rust used to describe `Error::description()` until it was deprecated via the metadata - since we're not adding that metadata here yet.

I would also change this sentence to be part of the doc detail and the summary line should just be "A NodeIdStore backed by a file".

File: cli/src/action/database/upgrade/mod.rs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+use super::Action;
+
+/// The overarching Action possibly containing multiple upgrade actions
+pub struct UpgradeAction;
+
+impl Action for UpgradeAction {
+    fn run<'a>(&mut self, arg_matches: Option<&ArgMatches<'a>>) -> Result<(), CliError> {
+        let state_dir = get_state_dir(arg_matches)?;
+        let database_uri = get_database_uri(arg_matches)?;
+        let connection_uri = database_uri.parse().map_err(|_| {
```
Comment (peterschwarz) [inline cli/src/action/database/upgrade/mod.rs:41]: Why not return a `ConnectionUri` from the `get_database_uri` function - it's already parsed as such to check connectivity.

File: cli/src/action/database/upgrade/mod.rs [hunk 4]
```diff
@@ -44,6 +46,11 @@ impl Action for UpgradeAction {
                 CliError::ActionError(format!("failed to initialized store factory: {}", err))
             })?;
 
+        #[cfg(feature = "node-id-upgrade")]
```
Comment (agunde406) [inline cli/src/action/database/upgrade/mod.rs:49]: Can you also update the man page for the `splinter upgrade` command

File: cli/src/action/database/upgrade/mod.rs (hunk 3 above)
Comment (Caleb-Hill) [reply cli/src/action/database/upgrade/mod.rs]: Updated to return a ConnectionUri

File: libsplinter/src/node_id/store/file/mod.rs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+//! A NodeIdStore backed by a file.
+
+use std::fs;
+use std::path::PathBuf;
+
+use super::NodeIdStore;
+use super::NodeIdStoreError;
+
+/// The 0.4 node_id file is soft-deprecated, this exists to help migrate the node_id.
```
Comment (peterschwarz) [inline libsplinter/src/node_id/store/file/mod.rs:23]: The summary line should be the same as the module doc. This line is more of a detail description.

Comment (Caleb-Hill) [reply libsplinter/src/node_id/store/file/mod.rs]: Updated the line to match the module doc

**Review by peterschwarz — APPROVED** [review]

File: cli/src/action/database/upgrade/node_id.rs [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    match (from.get_node_id(), to.get_node_id()) {
+        (Ok(Some(id)), Ok(None)) => to
+            .set_node_id(id)
+            .map_err(|e| CliError::ActionError(format!("{}", e)))
+            .map(|_| WarningEmited::No),
+        (Ok(Some(_)), Ok(Some(_))) => Err(CliError::ActionError(
+            "destination store already has node_id set".to_string(),
+        )),
+        (Ok(None), _) => {
+            warn!("node_id file is empty, skipping import");
```
Comment (agunde406) [inline cli/src/action/database/upgrade/node_id.rs:44]: Make this capital (and the the other warnings) to match the other logging that happens during the subcommand

Comment (agunde406) [reply cli/src/action/database/upgrade/node_id.rs:44]: I think we should have all of the skipping logs start like "Skipping ...."

File: cli/src/action/database/upgrade/yaml.rs [hunk 7]
```diff
@@ -101,9 +101,8 @@ pub fn import_yaml_state_to_database<P: Into<PathBuf>>(
     let proposals_location = state_dir.join(PROPOSALS_FILE);
 
     if !(circuits_location.exists() || proposals_location.exists()) {
-        return Err(CliError::ActionError(
-            "no yaml state files found to import".to_string(),
-        ));
+        warn!("no yaml state files found to import");
```
Comment (agunde406) [inline cli/src/action/database/upgrade/yaml.rs:104]: Maybe changed this to  "Skipping YAML circuit state import, no yaml state files found"

File: cli/src/action/database/upgrade/yaml.rs [hunk 8]
```diff
@@ -336,6 +336,7 @@ proposals:
     }
 
     #[test]
+    #[ignore]
```
Comment (agunde406) [inline cli/src/action/database/upgrade/yaml.rs:339]: I think this test should just be updated to verify that it return Ok(())

**Review by agunde406 — DISMISSED** [review]

**Review by peterschwarz — APPROVED** [review]

**Review by shannynalayna — APPROVED** [review]
